win32/src/winapi/bitmap.rs

```diff
@@ -52,6 +69,19 @@ impl BITMAPINFOHEADER {
     }
 }
 
+pub struct BitmapInfo {
+    pub width: usize,
+    pub height: usize,
+    pub stride: usize,
+    pub is_top_down: bool,
+    pub bit_count: u8,
+    pub compression: BI,
+    pub palette_entry_size: usize,
+    pub palette_size: usize,
+    pub palette: *const u8,
+    pub data: *const u8,
```

Comment:
I am a little worried about raw pointers here because the lifetimes aren't too clear.  It looks like from how this is used, these could both be &[u8] references, I think?

Reply:
This is my first time writing a large amount of Rust code, and I haven’t fully understood the concept of lifetimes yet, making it difficult to get the code to compile. Please help me to make the corrections.

Reply:
No problem, I'll take a look!

Reply:
https://github.com/evmar/retrowin32/commit/009b4c9efcb0ba2ec373e2dc1b0dc77f374c9f7c

`BitmapInfo<'a>` means "a BitmapInfo whose pointers last as long as lifetime 'a", and Rust is able to infer that the lifetime is long enough without any other annotations, so this one ended up pretty easy.  (If a function returned a BitmapInfo then the lifetime annotations might be more complicated because you'd need to show how the palette/data member pointers stay valid after returning.)